### ISSUE TITLE
Unknown personas silently bypass the reward-model prompt wrapper, mixing two incompatible input formats in one training set

### ISSUE THREAD

**Issue by dolceo** [hello-diana/MASCOT#2]
`src/train/train_reward_model.py:102`

```python
    def wrap_with_eval_prompt(example):
        persona_key = example.get("persona", "")
        if persona_key in EMPATHETIC_AGENTS:
            persona_spec = EMPATHETIC_AGENTS[persona_key]
        elif persona_key in AGENTS:
            persona_spec = AGENTS[persona_key]
        else:
            return example  # unknown persona, leave unchanged

        for col in ("chosen", "rejected"):
            ...
            example[col] = [{"role": "user", "content": eval_prompt}]
        return example
```

The `else` branch is a silent no-op. Rows that hit it keep their original multi-turn `chosen`/`rejected` message lists, while every other row is rewritten into a single synthetic user turn containing the persona rubric.

## How it triggers

Any row lacking a `persona` column — which is exactly what the multi-dataset path produces. `src/train/train_reward_model.py:85` back-fills missing non-core columns with empty strings before concatenation:

```python
        for i, ds in enumerate(datasets_list):
            for col in all_cols:
                if col not in ds.column_names and col not in core_cols:
                    datasets_list[i] = ds.add_column(col, [""] * len(ds))
```

So combine a dataset that has `persona` with one that does not, and every row of the second gets `persona == ""`, falls into the `else`, and trains unwrapped. Same for a persona key renamed in `config/persona_config.py` but not in the cached dataset on disk.

## Consequence

The comment at `src/train/train_reward_model.py:100` states the intent plainly — the wrapper exists so the RM "learns to score based on explicit persona/role/criteria (matching compute_reward at inference)". Rows that skip it teach the model to score a raw dialogue instead. At inference the wrapper is always applied, so those training examples are pure distribution mismatch. The log line at `src/train/train_reward_model.py:124` unconditionally announces success:

```python
    logger.info("Wrapped responses in evaluation prompt (persona/role/criteria)")
```

There is no counter, so a run where 100% of rows were skipped is indistinguishable in the logs from a run where 0% were.

## Expected

Count the skips and either raise or drop the rows:

```python
    n_before = len(train_dataset)
    train_dataset = train_dataset.filter(lambda ex: ex.get("persona") in AGENTS or ex.get("persona") in EMPATHETIC_AGENTS)
    if len(train_dataset) < n_before:
        logger.warning(f"dropped {n_before - len(train_dataset)} rows with unknown persona")
```

and fail hard if the drop rate exceeds a threshold. Silently training on a mixed format is the worst of the three options.

**Comment by Ahren09** [comment]
Thanks for the heads-up. We compared our released code with the development version and confirmed that the issue was introduced during our public release.

The fix follows your suggested shape:

1. **Filter before wrapping**: after the train/eval split, rows whose `persona` is not in `AGENTS`/`EMPATHETIC_AGENTS` are dropped from both splits, with a warning that logs the exact count and fraction dropped.
2. **Hard failure past a threshold**: if the dropped fraction exceeds `--max-unknown-persona-frac` (default 1%), training aborts with a `ValueError` pointing at the likely causes (a dataset in `--data-path` missing the persona column, or persona keys out of sync with `config/persona_config.py`).
3. **No silent path left**: the `else` branch in `wrap_with_eval_prompt` now raises instead of returning the example unchanged. It is unreachable after the filter.
